File: reference/sliding-window/1984-Minimum-Difference-Between-Highest-and-Lowest-of-K-Scores.go
func minimumDifference(nums []int, k int) int {
    sort.Ints(nums)
    right := k - 1
    ans := math.MaxInt32

    for left := 0; left < len(nums); left++ {
        if right < len(nums) {
            ans = min(nums[right] - nums[left], ans)
            right++
        }
    }
    return ans
}

File: reference/sliding-window/1984-Minimum-Difference-Between-Highest-and-Lowest-of-K-Scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        right = k - 1
        ans = float('inf')
        for left in range(len(nums)):
            if right < len(nums):
                ans = min(nums[right] - nums[left], ans)
                right += 1
        return ans